lib/clients/http_client.js

```diff
@@ -628,6 +629,15 @@ HttpClient.prototype._options = function (method, options) {
         opts.socketPath = this.socketPath;
     }
 
+    // Add the request path to the base path of the client
+    // Don't combine the request url and client path if they both
+    // have a slash
+    if (opts.path && this.url.path &&
+        opts.path !== '/' && this.url.path !== '/') {
+        // clean up the url path, stripping the last /
+        opts.path = utils.sanitizePath([this.url.path, opts.path].join('/'));
```

Comment:
Based on #692, I think it's safe to say that we probably don't want to strip trailing slashes. Make the users be explicit?